projects/net-snmp/snmp_parse_oid_fuzzer.c

```diff
@@ -37,46 +38,13 @@ int SecmodInMsg_CB(struct snmp_secmod_incoming_params *sp1) {
 }
 
 int LLVMFuzzerTestOneInput(const uint8_t *data, size_t size) {
-    // We need to have at least oen byte for our decider var.
-    if (size == 0) {
-        return 0;
-    }
-    const uint8_t decider = *data;
-    data += 1;
-    size -= 1;
-
-    char *new_str = malloc(size+1);
-    if (new_str == NULL){
-        return 0;
-    }
-    memcpy(new_str, data, size);
-    new_str[size] = '\0';
-
-    // This fuzzer hits multiple entrypoints, use the first byte of the fuzz
-    // data to decide which entrypoint. 
-    switch (decider % 3) {
-        case 0:  {
-            oid *root = malloc(MAX_OID_LEN * sizeof(oid));
-            size_t rootlen;
-            snmp_parse_oid(new_str, root, &rootlen);
-            free(root);
-            break;
-        }
-        case 1: {
-            oid *objid = malloc(MAX_OID_LEN * sizeof(oid));
-            size_t objidlen = MAX_OID_LEN;
-            read_objid(new_str, objid, &objidlen);
-            free(objid);
-            break;
-        }
-        case 2: {
-            unsigned char *new_val;
-            size_t new_val_len;
-            parse_octet_hint(new_str, new_str, &new_val, &new_val_len);
-            break;
-        }
-    }
+    oid *root = malloc(MAX_OID_LEN * sizeof(oid));
+    size_t rootlen;
+    char *input;
 
-    free(new_str);
+    input = strndup((const char *)data, size);
+    snmp_parse_oid((const char *)data, root, &rootlen);
```

Comment:
Here you should use `input` as first argument to `snmp_parse_oid` rather than `data`.

Reply:
Fixed.